stingray/crossspectrum.py

```diff
@@ -1179,11 +1207,14 @@ def coherence(self):
         unnorm_powers_avg_2 = self.pds2.power.real
 
         coh = num / (unnorm_powers_avg_1 * unnorm_powers_avg_2)
+        coh[~np.isfinite(coh)] = 0.0
 
         # Calculate uncertainty
         uncertainty = \
             (2 ** 0.5 * coh * (1 - coh)) / (np.abs(coh) * self.m ** 0.5)
 
+        uncertainty[coh == 0] = 0.0
```

Comment:
@dhuppenkothen, @abigailStev: do we need to use these values to fit coherence spectra? If so, setting the uncertainty to 0 might give wrong results? Wouldn't it be better to set these to 1 instead (since coherence cannot be more than 1)?

Reply:
Well, I don't know, since I don't really have experience fitting coherence spectra, but: 
* if the coherence is zero, then the uncertainty is NaN (because you divide by zero in the line above), 
* current tests assume that in this case, the uncertainty on the coherence is zero, so I added this line to make sure that's true. If this is actually the desired behaviour, you are better positioned to judge than me, and if not some of those tests need changing. :)

Reply:
Ok, so: looking at the original paper, actually, it looks like the coherence function should be calculated in a more complicated way, at least accounting for the different cases of low and high powers.
https://iopscience.iop.org/article/10.1086/310430/pdf
So, I'd say that we merge this and we open an issue to verify that we are using the correct coherence function. What do you think?